azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -424,14 +424,23 @@ private void fetchMetaStoreToken(final String userToProxyFQN,
             logger.info("Need to pre-fetch extra metastore tokens from hive.");
 
             // start to process the user inputs.
+            int extraHcatTokenCount = 0;
             for (final String thriftUrl : extraHcatLocations) {
               logger.info("Pre-fetching metastore token from : " + thriftUrl);
 
               hiveConf = new HiveConf();
               hiveConf.set(HiveConf.ConfVars.METASTOREURIS.varname, thriftUrl);
-              hcatToken =
-                  fetchHcatToken(userToProxyFQN, hiveConf, thriftUrl, logger);
-              cred.addToken(hcatToken.getService(), hcatToken);
+              try {
+                hcatToken =
+                    fetchHcatToken(userToProxyFQN, hiveConf, thriftUrl, logger);
+                cred.addToken(hcatToken.getService(), hcatToken);
+                ++extraHcatTokenCount;
+              } catch (Exception e) {
+                logger.error("Failed to fetch extra metastore tokens from : " + thriftUrl, e);
+              }
+            }
+            if (0 == extraHcatTokenCount) {
+              throw new HadoopSecurityManagerException("No extra metastore token can be fetched.");
```

Comment:
Is this fix only required for EXTRA_HCAT_LOCATION? We don't need similar fix for EXTRA_HCAT_CLUSTERS?

Reply:
I added fix for EXTRA_HCAT_CLUSTERS as well.